tests/combine_samples.C

```diff
@@ -0,0 +1,190 @@
+    }
+	
+    CAROM_VERIFY((snaps > 0) && (dim > 0));
+	
+    /*-- Load data from input files --*/
+    std::string generator_filename = "total";
+    std::unique_ptr<CAROM::BasisGenerator> static_basis_generator;
+    static_basis_generator.reset(new CAROM::BasisGenerator(
+                                     CAROM::Options(dim, snaps).setMaxBasisDimension(snaps), false,
+                                     generator_filename));
+
+    if (rank==0) std::cout << "Loading data from " << kind << std::endl;
+    for(const auto& sample_name: sample_names) {
+        static_basis_generator->loadSamples(sample_name, kind);
+    }
+
+    if (rank==0) std::cout << "Saving data uploaded as a snapshot matrix" << std::endl;
+    static_basis_generator->writeSnapshot();
+	
+    if (!subtract_mean && !subtract_offset) {
+        /*-- Compute SVD and save file --*/
+        if (rank==0) std::cout << "Computing SVD" << std::endl;
+        int rom_dim = static_basis_generator->getSpatialBasis()->numColumns();
+        if (rank==0) std::cout << "U ROM Dimension: " << rom_dim << std::endl;
+        static_basis_generator->endSamples();
+    }
+    else {
+    	/*-- load data from hdf5 file to find the mean and subtract it --*/
+    	if (rank==0) std::cout << "Reading snapshots" << std::endl;
+    	CAROM::Matrix *snapshots = (CAROM::Matrix*) static_basis_generator->getSnapshotMatrix();
+        
+    	int num_rows = snapshots->numRows();
+    	int num_cols = snapshots->numColumns();
+        CAROM::Vector *offset = new CAROM::Vector(num_rows, true);
+    	
+        if (subtract_mean) {
+    	    /*-- Find the mean per row and write to hdf5 --*/
+            if (rank==0) std::cout << "Subtracting mean" << std::endl;
+    	    for (int row = 0; row < num_rows; ++row) {
+    	        (*offset)(row) = 0;
+    	        for (int col = 0; col < num_cols; ++col) {
+    	            (*offset)(row) += snapshots->item(row,col);
+    	        }
+    	    }
+            offset->mult(1.0 / (double)num_cols, offset);
+		    offset->write("mean");
+    	}
+        else if (subtract_offset) {
+            if (rank==0) std::cout << "Subtracting offset from file: " << offset_file << std::endl;
+            offset->read(offset_file);
+        }
+
+        /*-- Subtract mean or offset from snapshot/bases matrix --*/
+    	for (int row = 0; row < num_rows; ++row) {
+    	    for (int col = 0; col < num_cols; ++col) {
+    	        snapshots->item(row,col) = snapshots->item(row,col) - (*offset)(row);
+    	    }
+    	}
+    	
+    	/*-- Load new samples with mean/offset subtracted --*/
+    	std::unique_ptr<CAROM::BasisGenerator> static_basis_generator2;
+    	static_basis_generator2.reset(new CAROM::BasisGenerator(
+    	    CAROM::Options(dim, snaps).setMaxBasisDimension(snaps), false,
+    	    generator_filename+"_sub"));
+    	
+    	for (int col = 0; col < num_cols; col++) {
+            CAROM::Vector *snap_cur = new CAROM::Vector(num_rows, true);
```

Comment:
This allocates `snap_cur` for every column, without deleting it. It should be allocated before this loop over `col`. Also, it can be `CAROM::Vector snap_cur(num_rows, true);` so it doesn't have to be deleted.